rewrite-maven/src/main/java/org/openrewrite/maven/RemoveDuplicateDependencies.java

```diff
@@ -0,0 +1,147 @@
+
+        @Override
+        public Xml.Tag visitTag(Xml.Tag tag, ExecutionContext ctx) {
+            if (isDependenciesTag()) {
+                ctx.putMessage("dependencies", new HashMap<DependencyKey, Xml.Tag>());
+            } else if (isManagedDependenciesTag()) {
+                ctx.putMessage("managedDependencies", new HashMap<DependencyKey, Xml.Tag>());
+            } else if (isDependencyTag()) {
+                Map<DependencyKey, Xml.Tag> dependencies = ctx.getMessage("dependencies");
+                ResolvedDependency resolvedDependency = findDependency(tag);
+                Xml.Tag existing = dependencies.putIfAbsent(DependencyKey.from(resolvedDependency), tag);
+                if (existing != null && existing != tag) {
+                    maybeUpdateModel();
+                    return null;
+                }
+            } else if (isManagedDependencyTag()) {
+                Map<DependencyKey, Xml.Tag> dependencies = ctx.getMessage("managedDependencies");
+                ResolvedManagedDependency resolvedDependency = findManagedDependency(tag);
+                Xml.Tag existing = dependencies.putIfAbsent(DependencyKey.from(resolvedDependency), tag);
+                if (existing != null && existing != tag) {
+                    maybeUpdateModel();
+                    return null;
+                }
+            }
+            return super.visitTag(tag, ctx);
+        }
+
+        private boolean isDependenciesTag() {
+            return DEPENDENCIES_MATCHER.matches(getCursor());
+        }
+
+        private boolean isManagedDependenciesTag() {
+            return MANAGED_DEPENDENCIES_MATCHER.matches(getCursor());
+        }
+
+        @Nullable
+        @Override
+        public ResolvedDependency findDependency(Xml.Tag tag) {
+            Map<Scope, List<ResolvedDependency>> dependencies = getResolutionResult().getDependencies();
+            Scope scope = tag.getChildValue("scope").map(Scope::fromName).orElse(Scope.Compile);
+            if (dependencies.containsKey(scope)) {
+                for (ResolvedDependency resolvedDependency : dependencies.get(scope)) {
+                    Dependency req = resolvedDependency.getRequested();
+                    String reqGroup = req.getGroupId();
+                    if ((reqGroup == null || reqGroup.equals(tag.getChildValue("groupId").orElse(null))) &&
+                            Objects.equals(req.getArtifactId(), tag.getChildValue("artifactId").orElse(null)) &&
+                            Objects.equals(Optional.ofNullable(req.getType()).orElse("jar"), tag.getChildValue("type").orElse("jar")) &&
+                            Objects.equals(req.getClassifier(), tag.getChildValue("classifier").orElse(null))) {
+                        return resolvedDependency;
+                    }
+                }
+            }
+            return null;
+        }
+    }
+
+    @Value
+    private static class DependencyKey {
```

Comment:
I think we will need to add scope to the dependency key. Sorry, it has taken a while to get feedback to you. 

I am going to try to find some time in the next couple of days to provide some changes to this PR. Thanks for your patience.

Reply:
Yes, I think that was my plan. I must have forgotten about this. Thanks for the input!